src/python/pants/pantsd/pants_daemon.py

```diff
@@ -76,34 +76,58 @@ class RuntimeFailure(Exception):
 
   class Factory(object):
     @classmethod
-    def create(cls, bootstrap_options=None):
+    def maybe_launch(cls, bootstrap_options=None):
+      """Creates and launches a daemon instance if one does not already exist.
+
+      :param Options bootstrap_options: The bootstrap options, if available.
+      :returns: The pailgun port number of the running pantsd instance.
+      :rtype: int
+      """
+      stub_pantsd = cls.create(bootstrap_options, full_init=False)
+      with stub_pantsd.lifecycle_lock:
+        if stub_pantsd.needs_restart(stub_pantsd.options_fingerprint):
+          # Once we determine we actually need to launch, recreate with full initialization.
+          pantsd = cls.create(bootstrap_options)
+          return pantsd.launch()
+        else:
+          return stub_pantsd.read_named_socket('pailgun', int)
+
+    @classmethod
+    def create(cls, bootstrap_options=None, full_init=True):
       """
       :param Options bootstrap_options: The bootstrap options, if available.
+      :param bool full_init: Whether or not to fully initialize the engine.
```

Comment:
"full" isn't really defined anywhere, so it would be good to either: 1) describe what "full" means and what will not work if `!full`, 2) split this into two methods that accomplish the two halves of the initialization (ie, the first returns a result that is optionally passed to the second).